auth/strategies.js

```diff
@@ -0,0 +1,55 @@
+const passport = require('passport');
+const {BasicStrategy} = require('passport-http');
+const {Strategy: JwtStrategy, ExtractJwt} = require('passport-jwt');
+
+const {User} = require('../users/models');
+
+const basicStrategy = new BasicStrategy((username, password, callback) => {
+  let user;
+  User
+    .findOne({username: username})
+    .then(_user => {
+      user = _user;
+      if (!user) {
+        return Promise.reject({
+          reason: 'LoginError',
+          message: 'Incorrect username',
```

Comment:
Do we want to give away that usernames are correct/incorrect? Alternatively, we could log on server which part is wrong, and only tell client that auth credentials are wrong.